test/gtest/parser_unroll.cpp

```diff
@@ -311,4 +311,12 @@ TEST_F(P4CParserUnroll, t1_Cond) {
     ASSERT_EQ(parsers.first->states.size(), parsers.second->states.size());
 }
 
+TEST_F(P4CParserUnroll, header_union) {
+    auto parsers =  loadExample("issue561-7-bmv2.p4");
+    ASSERT_TRUE(parsers.first);
+    ASSERT_TRUE(parsers.second);
+    // adding outOfBound state
```

Comment:
What does this mean? Why is the number of parser states one less?

Reply:
This is because the parser adds an outOfBound state, which is correct. If we look at parser-unroll-test1.p4 we see similar behavior, as far as I understand it is caused by hitting similar parts of the code "return new SymbolicException (node, P4 :: StandardExceptions :: StackOutOfBounds);"

Reply:
But the tests says `ASSERT_EQ(parsers.first->states.size(), parsers.second->states.size() - 1)`;.
What states are removed here and why is this test correct? Is it because you do expect that executing the test does not crash and you get the right amount of parser states?  
In general, the explanations you provide here are helpful and should also be part of the test itself as part of comments.

Reply:
There I removed the outOfBound state, similar to other tests. Well, I'll add a more extensive comment to the code to make it clearer what exactly I removed.